### PR TITLE
mscolab working locally without created ~/mss folder, fixes #900

### LINKED ISSUE #900: Fatal error when working locally without ~/mss folder
When ticking the "Work Locally" checkbox in mscolab without an existing ~/mss folder, this fatal error pops up.
Screenshot at 2021-05-05 10-18-10

### PR DESCRIPTION
**Pull request by ReimarBauer**

### CODE REVIEW

File: mslib/msui/mscolab.py [hunk 1]
```diff
@@ -74,7 +76,8 @@ class MSSMscolabWindow(QtWidgets.QMainWindow, ui.Ui_MSSMscolabWindow):
     identifier = None
     viewCloses = QtCore.pyqtSignal(name="viewCloses")
 
-    def __init__(self, parent=None, data_dir=mss_default.mss_dir, mscolab_server_url=mss_default.mscolab_server_url):
+    def __init__(self, parent=None, data_dir=config_loader(dataset="mss_dir"),
```
Comment (joernu76) [inline mslib/msui/mscolab.py]: Does this work? Timing wise? Always? I would prefer None here with access to config_loader in the function if it is None, similar to how one should handle dicts/lists.

File: mslib/msui/mscolab.py [hunk 2]
```diff
@@ -530,8 +552,10 @@ def handle_work_locally_toggle(self):
             if self.version_window is not None:
                 self.version_window.close()
             self.create_local_project_file()
-            self.local_ftml_file = fs.path.join(self.data_dir, 'local_mscolab_data',
```
Comment (joernu76) [inline mslib/msui/mscolab.py:533]: I do not understand, from the documentation of fs, why this change is necessary. It seems to be the same, but more complicated...?

File: mslib/_tests/test_utils.py [hunk 3]
```diff
@@ -121,7 +121,7 @@ def test_existing_empty_config_file(self):
         with fs.open_fs(MSS_CONFIG_PATH) as file_dir:
             file_content = file_dir.readtext("mss_settings.json")
         assert ":" not in file_content
-        config_file = fs.path.join(MSS_CONFIG_PATH, "mss_settings.json")
+        config_file = fs.path.combine(MSS_CONFIG_PATH, "mss_settings.json")
```
Comment (joernu76) [inline mslib/_tests/test_utils.py:124]: I understand that this is more efficient?

**Review by joernu76 — COMMENTED** [review]
Some questions to understand fs better. Seems fine for a hot fix.

File: mslib/_tests/test_utils.py (hunk 3 above)
Comment (ReimarBauer) [reply mslib/_tests/test_utils.py:124]: this keeps the double slash, while the other removes one.

Comment (joernu76) [reply mslib/_tests/test_utils.py:124]: There is nothing in the documentation for that.

This method only supports fs-paths.

fs can only deal with PyFilesystem paths.
Conversion works using

unsyspath and getsyspath methods.

https://pyfilesystem.readthedocs.io/en/latest/osfs.html

This should solve our "problem" with QT. But for the JSON-config we need some speculative conversion routine, because we may have either an OS or a fs-path.

I am getting doubts if this is even fine for a hotfix.

**Comment by ReimarBauer** [comment]
issue #901 will become a seperate PR, if it is not possible to fix in stable this configeditor I probably will deactivate it for windows until we have it 

This PR is about the "work locally" worflow in the msco window.

**Review by joernu76 — APPROVED** [review]
